swift/ql/lib/codeql/swift/elements/expr/MethodLookupExpr.qll

```diff
@@ -0,0 +1,31 @@
+private import codeql.swift.generated.expr.MethodLookupExpr
+private import codeql.swift.elements.expr.MethodLookupExprConstructor
+private import codeql.swift.elements.expr.DeclRefExpr
+private import codeql.swift.elements.expr.OtherConstructorDeclRefExpr
+private import codeql.swift.elements.decl.Decl
+private import codeql.swift.elements.decl.MethodDecl
+private import codeql.swift.generated.Raw
+private import codeql.swift.generated.Synth
+
+class MethodLookupExpr extends Generated::MethodLookupExpr {
+  override string toString() { result = "." + this.getMember().toString() }
+
+  override Expr getImmediateBase() {
+    result = Synth::convertExprFromRaw(this.getUnderlying().getBase())
+  }
+
+  override Decl getImmediateMember() {
+    result = this.getMethodRef().(DeclRefExpr).getDecl()
+    or
+    result = this.getMethodRef().(OtherConstructorDeclRefExpr).getConstructorDecl()
+  }
+
+  override Expr getImmediateMethodRef() {
+    result = Synth::convertExprFromRaw(this.getUnderlying().getFunction())
+  }
+
+  MethodDecl getMethod() { result = this.getMember() }
+
+  cached
+  private Raw::SelfApplyExpr getUnderlying() { this = Synth::TMethodLookupExpr(result) }
```

Comment:
`Synth::TMethodLookupExpr` is already cached, so I don't think we need to cache this predicate (unless DCA has told us otherwise)?

Reply:
It was already a cached predicate for some reason, but yes, I can take a look at removing that.

Reply:
Oh. Yeah, that's surprising to me. This should be a simple `SCAN` through a cached table.